prestoadmin/connector.py

```diff
@@ -37,11 +38,30 @@
 COULD_NOT_REMOVE = 'Could not remove connector'
 
 
-def deploy_files(filenames, local_dir, remote_dir):
+def deploy_files(filenames, local_dir, remote_dir, user_group, mode=0600):
+    missing_owner_code = 42
+    user, group = user_group.split(":")
     _LOGGER.info('Deploying configurations for ' + str(filenames))
     sudo('mkdir -p ' + remote_dir)
     for name in filenames:
-        put(os.path.join(local_dir, name), remote_dir, use_sudo=True)
+        files = put(os.path.join(local_dir, name), remote_dir, use_sudo=True,
+                    mode=mode)
+        for file in files:
+            with settings(warn_only=True):
+                command = \
+                    "getent passwd {user} >/dev/null || ( rm -f {file} ; " \
```

Comment:
can you add commas around the || for clarity? I'm assuming that this check means that if the user doesn't exist, we fail to move the connector there, but it took a little while to parse.